daemon/cc_mode/container.proto

```diff
@@ -0,0 +1,166 @@
+/*
+ * This file is part of trust|me
+ * Copyright(c) 2013 - 2021 Fraunhofer AISEC
+ * Fraunhofer-Gesellschaft zur Förderung der angewandten Forschung e.V.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2 (GPL 2), as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GPL 2 license for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, see <http://www.gnu.org/licenses/>
+ *
+ * The full GNU General Public License is included in this distribution in
+ * the file called "COPYING".
+ *
+ * Contact Information:
+ * Fraunhofer AISEC <trustme@aisec.fraunhofer.de>
+ */
+
+syntax = "proto2";
+
+option java_package = "de.fraunhofer.aisec.trustme";
+
+message ContainerImageSize {
+	required string image_name = 1; // virtual name of the image file in guestos
+	required uint64 image_size = 2; // size (bytes) of the image file
+	optional string image_file = 3; // name of alternat image file which overwrites image_name of guestos config
+}
+
+message ContainerVnetConfig {
+	required string if_name = 1; // name of virtual veth endpoint in container
+	required bool configure = 2; // should cmld configure the interface or leav it unconfigured
+	optional string if_rootns_name = 3; // name of virtual veth endpoint in rootns (will be autogenerated by cmld)
+	optional string if_mac = 4; // mac of virtual veth endpoint inside container (will be autogenerated)
+	// TODO Define configuration, for now just use hardcoded default config in c_net
+}
+
+message ContainerPnetConfig {
+	required string netif = 1; // name or mac of physical network intarface mapped to container
+	repeated string mac_filter = 2; // mac of allowed client devices on that netfif
+}
+
+enum ContainerUsbType {
+	GENERIC = 1; // generic USB device
+	TOKEN = 2; // hardware security token
+	PIN_ENTRY = 3;  // pin entry device (pin pad)
+}
+
+message ContainerUsbConfig {
+	required string id = 1; // vendor:product USB IDs
+	required string serial = 2; // USB SERIAL_SHORT
+	required bool assign = 3 [default = false]; // assign the device to the container
+	required ContainerUsbType type = 4 [default = GENERIC];
+}
+
+/**
+ * Type of container KVM|CONTAINER
+ */
+enum ContainerType {
+	CONTAINER = 1;
+	KVM = 2	;
+}
+
+enum ContainerTokenType {
+	NONE = 1; // container does not use a token
+	SOFT = 2; // container uses an OpenSSL softoken
```

Comment:
Do we want to keep the options {NONE,SOFT} in CC_Mode?

Reply:
yes, since c0 uses TokenType SOFT